app/src/main/java/com/example/weathertimeapp/mvp/model/WeatherTimeModel.kt

```diff
@@ -30,7 +35,25 @@ class WeatherTimeModel(private val citiesJson: String) : WeatherTimeContracts.Mo
         return stringList
     }
 
-    override fun createDataSet(): ArrayList<Weather> {
-        return DataSource.createDataSet()
+    override fun getForecastByCityId(id: Int): Observable<ForecastResponse> {
+        return Observable.create { subscriber ->
+            val queryHashMap = LinkedHashMap<String, String>()
+            queryHashMap[ID_CITY] = id.toString()
+            queryHashMap[APPID] = ForecastRequestGenerator.API_KEY
+            val callResponse = api.createService(WeatherTimeApi::class.java).getCityById(queryHashMap)
+            val response = callResponse.execute()
+            if (response.isSuccessful) {
+                response.body()?.city?.id?.let {
```

Comment:
Why you use a scope function If you do not use the reference?

Reply:
removed, added a line to see the response on the debugger

Reply:
ok, not problem